src/syft/grid/duet/webrtc_duet.py

```diff
@@ -241,76 +257,94 @@ async def pull(self) -> None:
     def send_offer(self, target_id: str) -> None:
         """Starts a new signaling process by creating a new
         offer message and pushing it to the Signaling Server."""
-
-        # Generates an offer request payload containing
-        # local network description data/metadata (IP, MAC, Mask, etc...)
-        payload = asyncio.run(self.connection._set_offer())
-
-        # Creates a PySyft's SignalingOfferMessage
-        signaling_offer = SignalingOfferMessage(
-            address=self.signaling_client.address,  # Target's address
-            payload=payload,  # Offer Payload
-            host_metadata=self.node.get_metadata_for_client(),  # Own Node Metadata
-            target_peer=target_id,
-            host_peer=self.signaling_client.duet_id,  # Own Node ID
-        )
-
-        # Enqueue it in push msg queue to be sent to the signaling server.
-        self._push_msg_queue.put_nowait(signaling_offer)
-
-        # Create/enqueue a new AnswerPullRequest in order to wait for signaling response.
-        self._pull_msg_queue.put_nowait(
-            AnswerPullRequestMessage(
-                address=self.signaling_client.address,
+        try:
+            # Generates an offer request payload containing
+            # local network description data/metadata (IP, MAC, Mask, etc...)
+            payload = asyncio.run(self.connection._set_offer())
+
+            # Creates a PySyft's SignalingOfferMessage
+            signaling_offer = SignalingOfferMessage(
+                address=self.signaling_client.address,  # Target's address
+                payload=payload,  # Offer Payload
+                host_metadata=self.node.get_metadata_for_client(),  # Own Node Metadata
                 target_peer=target_id,
-                host_peer=self.signaling_client.duet_id,
-                reply_to=self.signaling_client.address,
+                host_peer=self.signaling_client.duet_id,  # Own Node ID
+            )
+
+            # Enqueue it in push msg queue to be sent to the signaling server.
+            self._push_msg_queue.put_nowait(signaling_offer)
+
+            # Create/enqueue a new AnswerPullRequest in order to wait for signaling response.
+            self._pull_msg_queue.put_nowait(
+                AnswerPullRequestMessage(
+                    address=self.signaling_client.address,
+                    target_peer=target_id,
+                    host_peer=self.signaling_client.duet_id,
+                    reply_to=self.signaling_client.address,
+                )
             )
-        )
+        except Exception as e:
+            log = f"Got an exception in Duet send_offer. {e}"
```

Comment:
NITPICK: For logging it is better to use a ```lazy``` string evaluation technique. Like
```logger.info("Something happened %s", var_string)```
^^^ In this scenario if the logger level is above ```info``` this would not be evaluated.

Using ```f-strings``` they would always be evaluated.

^^ This is premature optimization, but I think some flake flags or black might complain about this.

Reply:
Unfortunately, I dont think the logging library acts like print by supporting default printf string formatting like that.
```
var_string = "test"
sy.logger.info("Something happened %s", var_string)
# 2020-11-03 12:53:11.089 | INFO     | __main__:<module>:2 - Something happened %s
```

It does support the sort of .format syntax:
```
sy.logger.info("Something happened {x}", x=var_string)
sy.logger.info("Something happened {}", var_string)
```
So I guess that might be better?

Alternatively they have their own lazy, but it requires wrapping things in functions / lambdas.
https://github.com/Delgan/loguru#lazy-evaluation-of-expensive-functions
```
sy.logger.opt(lazy=True).critical("Something happened {}", lambda: var_string)
```

That starts to make the logging more verbose, so we could make a helper that goes in our global utils without circular import issues to make this a bit nicer, but it feels like this needs a bit more thought. I will add it to the new logging task.